### PR TITLE
check_by_ssh: Add "-U" flag (#1123).

### PR DESCRIPTION
**Pull request by archiecobbs**
This causes a 255 exit value from ssh(1), which indicates a connection failure,
to return UNKNOWN instead of CRITICAL; similar to check_nrpe's "-u" flag.

### CODE REVIEW

**Comment by RincewindsHat** [comment]
Resolves #1123 

Thank you very much for the PR.
I tried it and it seems, that it will change the behaviour only if `stderr` is ignored, since `ssh (1)` will output a message which will cause a `STATE_UNKNOWN` if `skip_stderr` is not present.
Seems like a valid use case anyway.
I would like one small change though, could you use to the `bool` type for the `unknown_timeout` variable to make the semantics more obvious?

File: plugins/check_by_ssh.c [hunk 1]
```diff
@@ -50,6 +50,7 @@ unsigned int services = 0;
 int skip_stdout = 0;
 int skip_stderr = 0;
 int warn_on_stderr = 0;
+int unknown_timeout = 0;
```
Comment (RincewindsHat) [inline plugins/check_by_ssh.c:53]: Please change the type to `bool` to make the semantics clearer

**Review by RincewindsHat — CHANGES_REQUESTED** [review]

**Comment by archiecobbs** [comment]
> it will change the behaviour only if stderr is ignored, since ssh (1) will output a message which will cause a STATE_UNKNOWN if skip_stderr is not present.

Not sure what you're talking about. The added patch happens on [line 105](https://github.com/archiecobbs/monitoring-plugins/blob/ca4b657381abdc760b8a54f74b8983877f506be9/plugins/check_by_ssh.c#L105) which is prior to [line 125](https://github.com/archiecobbs/monitoring-plugins/blob/ca4b657381abdc760b8a54f74b8983877f506be9/plugins/check_by_ssh.c#L125) where `skip_stderr` takes effect.

> I would like one small change though, could you use to the bool type for the unknown_timeout variable to make the semantics more obvious?

Good idea - thanks. Done.

**Review by RincewindsHat — APPROVED** [review]

**Comment by RincewindsHat** [comment]
Looks good to me. Thank you very much

**Pull request merged**